packages/p2p/src/components/verification/__tests__/verification.spec.tsx

```diff
@@ -0,0 +1,162 @@
+import React from 'react';
+import { screen, render } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { isDesktop } from '@deriv/shared';
+import { useModalManagerContext } from 'Components/modal-manager/modal-manager-context';
+import { useStores } from 'Stores/index';
+import Verification from '../verification';
+
+jest.mock('@deriv/shared', () => ({
+    ...jest.requireActual('@deriv/shared'),
+    isDesktop: jest.fn(() => false),
+}));
+
+const mock_modal_manager = {
+    isCurrentModal: false,
+    showModal: jest.fn(),
+    hideModal: jest.fn(),
+};
+
+jest.mock('Components/modal-manager/modal-manager-context', () => ({
+    ...jest.requireActual('Components/modal-manager/modal-manager-context'),
+    useModalManagerContext: jest.fn(() => mock_modal_manager as unknown as ReturnType<typeof useModalManagerContext>),
```

Comment:
i dont know if its better, just suggestion

```suggestion
    useModalManagerContext: jest.fn(() => ({
        ....mock_modal_manager
    })),
```

Reply:
https://github.com/binary-com/deriv-api-docs/blob/bb56c1442b8c3d16d510af133195bcbe77d55298/src/features/dashboard/components/AppsTable/__tests__/apps-table.test.tsx#LL9C1-L12C3

Reply:
we will use type safety by doing this

Reply:
we will use type safety by doing this @farhan-nurzi-deriv

Reply:
@farhan-nurzi-deriv please checkout the above example
it's better to use it like this and it's much cleaner 
the way you suggested will cause loosing type safety

Reply:
It will work like this too right? Because we need to cast the function return type isn't it

![image](https://github.com/farrah-deriv/deriv-app/assets/125247833/07dd3c75-28d0-456e-b5cf-cd9eaeae868e)